### ISSUE TITLE
support for long arrays

### ISSUE THREAD

**Issue by karasikov** [mklarqvist/libalgebra#5]
Dear Marcus,

Is it possible to change the length types to `uint64_t` so that the algorithms work for very long arrays as well (~2^35 of 8-byte words)?

I found another library https://github.com/kimwalisch/libpopcnt, where they use uint64_t, but unfortunately, it doesn't have your nice functions for computing intersection and union.

**Comment by mklarqvist** [comment]
Although the function interface is `f(const uint8_t* data, size_t size)` you are free to pass any primitive type as input by casting. For example, for 64-bit values you can call it like this:\
`f((const uint64_t*)data, size*sizeof(uint64_t))`. Internally, all computations are performed on 64-bit words.

**Comment by mklarqvist** [comment]
Importantly, this is true for all subroutines in this library with the exception of the pospopcnt family of operations which are limited to 16-bit input words.

**Comment by karasikov** [comment]
Yes, I wanted to ask about that... https://github.com/mklarqvist/libalgebra/blob/master/libalgebra.h#L567

But there are `int` types in other functions too, for instance in https://github.com/mklarqvist/libalgebra/blob/master/libalgebra.h#L1439,
where `int i` is compared to the `size_t n_ints`, potentially very large.
